ledger-core/virtual/integration/sender_test.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package integration
+
+import (
+	"context"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/gojuno/minimock/v3"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/insolar/assured-ledger/ledger-core/insolar/jet"
+	"github.com/insolar/assured-ledger/ledger-core/insolar/node"
+	"github.com/insolar/assured-ledger/ledger-core/insolar/payload"
+	"github.com/insolar/assured-ledger/ledger-core/instrumentation/inslogger"
+	"github.com/insolar/assured-ledger/ledger-core/pulse"
+	"github.com/insolar/assured-ledger/ledger-core/reference"
+	"github.com/insolar/assured-ledger/ledger-core/virtual/authentication"
+	"github.com/insolar/assured-ledger/ledger-core/virtual/integration/utils"
+)
+
+var messagesWithoutToken = []struct {
+	name string
+	msg  interface{}
+}{
+	{
+		name: "VCallRequest",
+		msg:  &payload.VCallRequest{},
+	},
+	{
+		name: "VCallResult",
+		msg:  &payload.VCallResult{},
+	},
+	{
+		name: "VStateRequest",
+		msg:  &payload.VStateRequest{},
+	},
+	{
+		name: "VStateReport",
+		msg:  &payload.VStateReport{},
+	},
+	{
+		name: "VDelegatedCallRequest",
+		msg:  &payload.VDelegatedCallRequest{},
+	},
+	{
+		name: "VDelegatedCallResponse",
+		msg:  &payload.VDelegatedCallResponse{},
+	},
+	{
+		name: "VFindCallRequest",
+		msg:  &payload.VFindCallRequest{},
+	},
+	{
+		name: "VFindCallResponse",
+		msg:  &payload.VFindCallResponse{},
+	},
+}
+
+func TestSender_EqualExpectedVE(t *testing.T) {
+	testCases := []struct {
+		name                    string
+		caseId                  string
+		senderIsEqualExpectedVE bool
+	}{
+		{"Sender is equal expectedVE", "C5188", true},
+		{"Sender is not equal expectedVE", "C5196", false},
+	}
+
+	for _, cases := range testCases {
+		t.Run(cases.name, func(t *testing.T) {
+			t.Log(cases.caseId)
+			for _, testMsg := range messagesWithoutToken {
+				t.Run(testMsg.name, func(t *testing.T) {
+
+					mc := minimock.NewController(t)
+
+					server, ctx := utils.NewUninitializedServerWithErrorFilter(nil, t, func(s string) bool {
```

Comment:
But why you don't set error handler here?

Reply:
there should be check that only expected errors are skipped